client/extensions/woocommerce/app/settings/taxes/index.js

```diff
@@ -0,0 +1,203 @@
+import classNames from 'classnames';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import ActionHeader from 'woocommerce/components/action-header';
+import {
+	areSettingsGeneralLoading,
+	areTaxCalculationsEnabled
+} from 'woocommerce/state/sites/settings/general/selectors';
+import {
+	areTaxSettingsLoading,
+	getPricesIncludeTax,
+	getShippingIsTaxFree
+} from 'woocommerce/state/sites/settings/tax/selectors';
+import Button from 'components/button';
+import {
+	fetchSettingsGeneral,
+	updateTaxesEnabledSetting
+} from 'woocommerce/state/sites/settings/general/actions';
+import {
+	fetchTaxSettings,
+	updateTaxSettings
+} from 'woocommerce/state/sites/settings/tax/actions';
+import { getLink } from 'woocommerce/lib/nav-utils';
+import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
+import Main from 'components/main';
+import { successNotice, errorNotice } from 'state/notices/actions';
+import TaxesNexus from './taxes-nexus';
+import TaxesOptions from './taxes-options';
+import TaxesRates from './taxes-rates';
+
+class SettingsTaxes extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
+			isSaving: false,
+			pricesIncludeTaxes: true,
+			shippingIsTaxable: true,
+			taxesEnabled: true,
+			userBeganEditing: false,
+		};
+	}
+
+	static propTypes = {
+		site: PropTypes.shape( {
+			slug: PropTypes.string,
+		} ),
+		className: PropTypes.string,
+	};
+
+	componentDidMount = () => {
+		const { site } = this.props;
+
+		if ( site && site.ID ) {
+			this.props.fetchSettingsGeneral( site.ID );
+			this.props.fetchTaxSettings( site.ID );
+		}
+	}
+
+	componentWillReceiveProps = ( newProps ) => {
+		if ( ! this.state.userBeganEditing ) {
+			this.setState( {
+				pricesIncludeTaxes: newProps.pricesIncludeTaxes,
+				shippingIsTaxable: newProps.shippingIsTaxable,
+				taxesEnabled: newProps.taxesEnabled,
+			} );
+		}
+	}
+
+	onEnabledChange = () => {
+		this.setState( { taxesEnabled: ! this.state.taxesEnabled, userBeganEditing: true } );
+	}
+
+	onCheckboxChange = ( event ) => {
+		const option = event.target.name;
+		const value = event.target.checked;
+		this.setState( { [ option ]: value, userBeganEditing: true } );
+	}
+
+	pageHasChanges = () => {
+		return this.state.userBeganEditing;
+	}
+
+	onSave = ( event ) => {
+		const { site, translate } = this.props;
+
+		event.preventDefault();
+		this.setState( { isSaving: true } );
+
+		const onSuccess = () => {
+			this.setState( { isSaving: false, userBeganEditing: false } );
+			return successNotice( translate( 'Settings updated successfully.' ) );
+		};
+
+		const onFailure = () => {
+			this.setState( { isSaving: false } );
+			return errorNotice( translate( 'There was a problem saving your changes. Please try again.' ) );
+		};
+
+		// TODO - chain these
+
+		this.props.updateTaxesEnabledSetting(
+			site.ID,
+			this.state.taxesEnabled,
+			null,
```

Comment:
Agreed. Fixed in https://github.com/Automattic/wp-calypso/pull/15541/commits/17f50964f1f6156641032d53a6b96785ad2895a0